stdlib/3/typing.pyi

```diff
@@ -585,17 +596,33 @@ class Pattern(Generic[AnyStr]):
 
 # Functions
 
+_get_type_hints_obj_allowed_types = (
+    FunctionType,
+    BuiltinFunctionType,
+    MethodType,
+    ModuleType,
+)
+
+if sys.version >= (3, 7):
+    _get_type_hints_obj_allowed_types += (
+        WrapperDescriptorType,
+        MethodWrapperType,
+        MethodDescriptorType,
+    )
```

Comment:
I doubt that works with all type checkers. Just have an `else` branch and list all types in all branches. Also, you need to use a `Union` here:

```python
if sys.version >= (3, 7):
    _get_type_hints_obj_allowed_types = Union[...]
else:
    _get_type_hints_obj_allowed_types = Union[...]
```